runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -897,6 +900,104 @@ J9::TransformUtil::foldStaticFinalFieldAssumingProtection(TR::Compilation *comp,
    return false;
    }
 
+static bool
+classHasFinalPutstaticOutsideClinit(TR::Compilation *comp, TR_OpaqueClassBlock *clazz)
+   {
+   TR_J9VMBase *fej9 = comp->fej9();
+   J9VMThread *vmThread = fej9->vmThread();
+
+   J9ROMClass *romClass = TR::Compiler->cls.romClassOf(clazz);
+   TR::Region &stackRegion = comp->trMemory()->currentStackRegion();
+
+   // Get all of the final fields declared by clazz from the VM.
+   J9Class *j9c = TR::Compiler->cls.convertClassOffsetToClassPtr(clazz);
+   uintptr_t numStaticFields =
+      fej9->_vmFunctionTable->getStaticFields(vmThread, romClass, NULL);
```

Comment:
I did run this myself using JITServer, and I didn't hit any problem with this scan. The only problem I saw was with setting the flag on the class, which I had anticipated, and which worked with the new message

I also ran some of my routine testing with JITServer, and AFAICT all of the failures were known issues

I think it would still make sense to run JITServer PR builds